### PR TITLE
spacebar: new port

### PR DESCRIPTION
**Pull request by bashu**
#### Description

A minimal status bar for macOS.

###### Type(s)

- [x] enhancement

###### Tested on

macOS 12.6.1 21G217 x86_64
Xcode 14.1 14B47b

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by bashu** [comment]
I need to rework it...

**Comment by bashu** [comment]
Nevermind, I messed up my own personal config, but "spacebar" just works out-of-the box with it's own default settings.

File: sysutils/spacebar/Portfile [hunk 1]
```diff
@@ -0,0 +1,72 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem              1.0
+PortGroup               github 1.0
+PortGroup               makefile 1.0
+PortGroup               xcodeversion 1.0
+
+github.setup            cmacrae spacebar 1.4.0 v
+
```
Comment (reneeotten) [inline sysutils/spacebar/Portfile:9]: if upstream provides "releases" you would ideally set `github.tarball_from releases`; if not the next preferred option is `github.tarball_from archive`. You will need to update the `checksums` when making that change.

Comment (reneeotten) [inline sysutils/spacebar/Portfile:9]: if upstream provides "releases" you would ideally set `github.tarball_from releases`; if not the next preferred option is `github.tarball_from archive`. You will need to update the `checksums` when making that change.

File: sysutils/spacebar/Portfile [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+}
+
+set minxcodever 8.0
+if {([vercmp $xcodeversion ${minxcodever}] < 0)} {
+    known_fail yes
+    pre-fetch {
+        ui_error "${name} ${version} requires Xcode ${minxcodever} or greater to build."
+        return -code error "incompatible Xcode version"
+    }
+}
```
Comment (reneeotten) [inline sysutils/spacebar/Portfile:61]: I don't think this is how the test is supposed to work, but am not 100% sure. You do add the `xcodeversion` PortGroup at the top of the file but are not using it... Are you not supposed to just use `minimum_xcodeversions   {darwin_major minimum_xcodeversion}` (see [here](https://github.com/macports/macports-ports/blob/master/_resources/port1.0/group/xcodeversion-1.0.tcl))?

**Review by reneeotten — CHANGES_REQUESTED** [review]
thanks @bashu for the submission - a few comments below

File: sysutils/spacebar/Portfile (hunk 1 above)
Comment (bashu) [reply sysutils/spacebar/Portfile:12]: fixed

File: sysutils/spacebar/Portfile (hunk 2 above)
Comment (bashu) [reply sysutils/spacebar/Portfile]: fixed

**Pull request merged**